### ISSUE TITLE
Ship API: SurePost not listed as a valid service + No information anywhere about production environment access.

### ISSUE THREAD

**Issue by AlexAtPeerless** [UPS-API/api-documentation#93]
Hello, I have 2 questions:
-Both the OpenAPI spec and the online documentation do not list SurePost as a valid service for the Ship API. What's up with that? we absolutely need to be able to keep ship using that service.
-There is no information I could find about gaining access to the RESTful APIs production environment, do we just use the production URL with the same app OAuth clientId/secret to get a token and that's all? (I'm asking because with the older APIs there was this very irritating and time-consuming approval process, which if it still exists for the RESTful APIs definitely should NOT be unannounced as it is now...)

Thanks.

**Comment by ataylorkato** [comment]
I can at least answer the second one - UPS doesn't require you to go through any sort of approval process to access production with the new APIs. You can just swap to the production URL whenever you're ready. The same client_id/client_secret/bearer tokens will work against CIE and production.

**Comment by ataylorkato** [comment]
For the first question there's this response from @UPSRahul on another issue which admittedly isn't very helpful in an immediate sense. https://github.com/UPS-API/api-documentation/issues/71

**Comment by UPSRahul** [comment]
Use of the UPS Surepost product requires a signed agreement that ensures the customer integration of Surepost complies with UPS guidelines.  Under the process today once a signed agreement is in place the Surepost Rate and Ship API supplements are automatically distributed.  

Existing customers that are updating their interface from soap or XML to RESTful can contact the UPS Help Desk https://developer.ups.com/contactus to verify their contract via their account number and will soon be able to gain the documents. The Help Desk will be necessary as we do not want to share PII on the GitHub.

To access the production environment, you will need to use the production (onlinetools) URL along with production credentials which can be created on the UPS Developer Portal. Please reference the credentials piece of this page for instructions: [Getting Started with UPS APIs | UPS](https://developer.ups.com/get-started?loc=en_US)"

**Comment by deckberg-quantumdigital** [comment]
We contacted the help desk and they told us we needed to have our sales rep get us that info.  We've been working with our sales rep the last two weeks and today they finally sent us the XML documentation that we already received from them in 2020.  How do we get the RESTful SurePost info.  This is crazy.

**Comment by deckberg-quantumdigital** [comment]
Ok, our sales rep just got back to us saying this:

"Good morning, Tyler:
I questioned that as well with the developer person. This is the email response I received back from them:
There is NO RESTful Guide available for SurePost Documentation.
There are NO changes for SurePost guides. Since SurePost guides does NOT contain any schema."

**Comment by deckberg-quantumdigital** [comment]
Success!  After a bunch of trial and error it turns out that is correct that there is no new documentation and it just works.

**Comment by AlexAtPeerless** [comment]
Hello @deckberg-quantumdigital , I have been going through the same hoops and got the 2020 XML API documentation too last week, but haven't reached the step yet where they say there is no supplement for RESTful API after I complained they sent me the old documentation for the XML API we are going away from... 
They used to have a different XML schema for SurePost so there were some SurePost specific fields which you are telling us don't exist with RESTful API. 
So you just use same service code 93 and respect the same business rules and it just works? 
In his answer @UPSRahul seemed to think the SurePost supplement did exist for RESTful so this is a bit confusing (and time-wasting trying to get documentation that doesn't exist).

**Comment by deckberg-quantumdigital** [comment]
yes.  of course you have to do Rate instead of Shop and you can't request time-in-transit info.  So you are forced to do TWO API calls and then merge your results of Shop and Rate(93)+estimated TNT.  Very inefficient in my opinion.  It would be much better if UPS just looked at our account number and would know that we have a signed agreement in place and could return Service/Code=93 when ONE Shop request.  Hopefully, @UPSRahul is able to pass this along to the decision makers ;)

We also discovered this great find.  https://developer.ups.com/api/reference/steppingstone/endpoints?loc=en_US
We can indeed still use the SOAP/XML requests/response until they are removed at a time later than June 2024.  You just have to change the endpoint url slightly and pass your Authorization header with your post instead of your access credentials.

**Comment by ataylorkato** [comment]
You've got to be kidding. Is this documented or discussed anywhere else? Is there an EOL for this "stepping stone"?

**Comment by AlexAtPeerless** [comment]
@deckberg-quantumdigital thanks for the info.
I can't believe we actually get more knowledgeable answers from other users than from UPS support...
I'm positive the stepping-stone endpoints were not published not that long ago, and UPS tech support litteraly told me in december XML API cannot work with OAuth.

**Comment by nightcoding2021** [comment]
I've posted the instructions in another thread.

The JSON migration was a nightmare and although I haven't finished, I'll probably see it through.  Anyone though who is using all the intricacies of the XML API - that would be a mammoth task.  Even in my scenario, I'd slowly built up the XML integration with more and more complexity over the last two decades.
